### PR TITLE
feat(hooks): add useCollectionQuery and useIsMounted [JOB-32207]

### PR DESCRIPTION
**Pull request by jlapstra**
## Motivations

Both Jobber Online and Jobber Mobile use useCollectionQuery for 2WSMS. The version in Jobber Online, however did not have tests as it was implemented in Jobber Online during a big bug fix. It was always the intention to move useCollectionQuery to Atlantis, no time like now.

## Changes

To make useCollectionQuery work, we needed to also move over useIsMounted.
Tests have been written.

### Added

Both hooks can now be updated from a shared component in Atlantis

### Changed

The functionality should not change.

### Deprecated

NA

### Removed

NA

### Fixed

NA

### Security

NA

## Testing

Tests should work. Integrations with Jobber Online and Mobile should be tested completely.

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

File: packages/hooks/src/useIsMounted/useIsMounted.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { useLayoutEffect, useRef } from "react";
+
+/**
+ * Why does this work?
```
Comment (jlapstra) [inline packages/hooks/src/useIsMounted/useIsMounted.ts:4]: Should this go in an MDX file? @andrewjdmccann

Comment (andrewjdmccann) [reply packages/hooks/src/useIsMounted/useIsMounted.ts:4]: It should but I think it should also be in this file.

File: packages/hooks/src/useCollectionQuery/__mocks__/queries.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
+import gql from "graphql-tag";
+
+export const LIST_QUERY = gql`
```
Comment (jlapstra) [inline packages/hooks/src/useCollectionQuery/__mocks__/queries.ts:3]: is this actually the `list query` or the `conversation query`? @andrewjdmccann

**Comment by jlapstra** [comment]
@andrewjdmccann This looks good! I do believe it is the `conversation query / conversation query type` that we want here, instead of the `list query / list query type`.

File: packages/hooks/src/useCollectionQuery/__mocks__/queries.ts (hunk 2 above)
Comment (andrewjdmccann) [reply packages/hooks/src/useCollectionQuery/__mocks__/queries.ts:3]: This was a very intentional change. While this particular implementation is a Conversation query, the "interface" is that all we really need is for it to return a list. We don't care about anything else.

**Comment by jlapstra** [comment]
LGTM 👍

**Pull request merged**